packages/perseus-editor/src/components/__tests__/util.test.ts

```diff
@@ -276,3 +277,34 @@ describe("getSinusoidEquation", () => {
         expect(equation).toBe(expected);
     });
 });
+
+describe("getQuadraticEquation", () => {
+    test.each`
+        point1     | point2     | point3    | expected
+        ${[-5, 5]} | ${[0, -5]} | ${[5, 5]} | ${"y = 0.400x^2 + 0.000x + -5.000"}
+        ${[-5, 5]} | ${[0, 5]}  | ${[5, 5]} | ${"y = 0.000x^2 + 0.000x + 5.000"}
+        ${[-9, 9]} | ${[-7, 7]} | ${[9, 9]} | ${"y = 0.063x^2 + 0.000x + 3.938"}
+        ${[-9, 4]} | ${[-7, 7]} | ${[9, 9]} | ${"y = -0.076x^2 + 0.278x + 12.688"}
+        ${[-1, 0]} | ${[0, 1]}  | ${[1, 2]} | ${"y = 0.000x^2 + 1.000x + 1.000"}
+    `(
+        "should return the correct equation",
+        ({point1, point2, point3, expected}) => {
+            // Act
+            const equation = getQuadraticEquation([point1, point2, point3]);
+
+            expect(equation).toBe(expected);
+        },
+    );
+
+    test.each`
+        point1     | point2      | point3
+        ${[-5, 5]} | ${[-5, -5]} | ${[5, 5]}
+        ${[-5, 5]} | ${[0, -5]}  | ${[-5, 5]}
+        ${[-5, 5]} | ${[0, 5]}   | ${[0, 5]}
+    `("should return division by zero error", ({point1, point2, point3}) => {
+        // Act
+        const equation = getQuadraticEquation([point1, point2, point3]);
+
+        expect(equation).toBe("Division by zero error");
```

Comment:
This is an interesting case! Thanks for testing it.

I am curious: how does the graph preview render in this case? What happens to the graph when the start coords aren't valid?

Reply:
It just shows all the points where they would be, but the quadratic graph doesn't go through the points. The equation box says "Division by zero error"

<img width="483" alt="Screenshot 2024-07-31 at 3 43 19 PM" src="https://github.com/user-attachments/assets/38080ee8-d143-4bee-9e70-e6f526badb98">
<img width="377" alt="Screenshot 2024-07-31 at 3 43 33 PM" src="https://github.com/user-attachments/assets/636aa159-35b4-4c64-b802-162127eb1887">

Reply:
I'm not too worried about it because I think this is within the "trust the content authors" territory